assets/js/amazon-wc-checkout.js

```diff
@@ -1,40 +1,99 @@
-/*global amazon_payments_advanced, amazon */
+/*global amazon_payments_advanced, amazon, wc_checkout_params */
 ( function( $ ) {
 	$( function() {
 		var button_count = 0;
 		var button_id = '#pay_with_amazon';
-		function renderButton() {
+		var classic_button_id = '#classic_pay_with_amazon';
+		var amzCreateCheckoutConfig = null;
+		$( 'form.checkout' ).on( 'checkout_place_order_success', function( e, result ) {
+			if ( 'undefined' !== typeof result.amzCreateCheckoutParams && $( classic_button_id ).length > 0 ) {
+				amzCreateCheckoutConfig = result.amzCreateCheckoutParams;
+				renderButton( classic_button_id, 'classic' );
+				$( classic_button_id ).trigger( 'click' );
```

Comment:
Consider replacing with `amazon.Pay.initCheckout();` if possible.

Reply:
did a few tests but without success, will keep trying and ill keep you posted.

Reply:
Managed to make it work using initCheckout instead of trigger('click') but only when storing the amazon.Pay.renderButton return's value in a variable belonging to a bigger scope than would logically be required.

Reply:
Wouldn't it make sense to have `renderButton()` return the button?

Reply:
Done @ksere !